nestkernel/vose.cpp

```diff
@@ -38,30 +38,34 @@ Vose::Vose( std::vector< double > dist )
 
   // We accept distributions that do not sum to 1.
   double sum = 0.0;
-  for ( std::vector< double >::iterator it = dist.begin(); it != dist.end(); ++it )
+  for ( std::vector< double >::const_iterator it = dist.begin(); it != dist.end(); ++it )
```

Comment:
Can this be replaced with `auto it`? Or maybe even a range-based loop?

Reply:
In principle, yes, but in view of your comment above I'd want to minimise changes here.